indexservice/indexer.go

```diff
@@ -19,56 +20,36 @@ import (
 	"github.com/iotexproject/iotex-core/pkg/hash"
 )
 
+// If you want to add a new index table, please:
+// 1. add a index unique identifier below
+// 2. add this index unique identifier to one of table list
+const (
+	IndexTransfer  = "transfer"
+	IndexVote      = "vote"
+	IndexExecution = "execution"
+	IndexAction    = "action"
+	IndexReceipt   = "receipt"
+)
+
+var (
+	// BlockByIndexList store list of BlockByIndex tables
+	BlockByIndexList = []string{IndexTransfer, IndexVote, IndexExecution, IndexAction, IndexReceipt}
```

Comment:
this could be a config in indexer, which defines all the indexes this indexer wants to build

Reply:
I move it to config.go .   I have to move index Indentifier as well because other wise it will introduce cycle dependency. But after I move both of them to config, it start to become a little confusing in naming, because now we need to call config.IndexVote instead of indexservice.IndexVote. But I understand the motivation is here is make those table list configurable